src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -0,0 +1,116 @@
+package io.iohk.ethereum.network
+
+import java.net.URI
+
+import io.iohk.ethereum.network.p2p.messages.WireProtocol._
+import io.iohk.ethereum.rlp.RLPEncoder
+
+import scala.concurrent.duration._
+
+import akka.actor._
+import akka.util.ByteString
+import io.iohk.ethereum.network.RLPxConnectionHandler.MessageReceived
+import io.iohk.ethereum.network.p2p._
+
+class PeerActor(nodeInfo: NodeInfo) extends Actor with ActorLogging {
+
+  import PeerActor._
+  import context.{dispatcher, system}
+
+  val P2pVersion = 4
+  val EthProtocolVersion = 63.toByte
+
+  val peerId = self.path.name
+
+  override def receive: Receive = waitingForInitialCommand
+
+  def waitingForInitialCommand: Receive = {
+    case HandleConnection(connection) =>
+      val rlpxConnection = createRlpxConnection()
+      rlpxConnection ! RLPxConnectionHandler.HandleConnection(connection)
+      context become waitingForConnectionResult(rlpxConnection)
+
+    case ConnectTo(uri) =>
+      val rlpxConnection = createRlpxConnection()
+      rlpxConnection ! RLPxConnectionHandler.ConnectTo(uri)
+      context become waitingForConnectionResult(rlpxConnection)
+  }
+
+  def createRlpxConnection(): ActorRef = {
+    val rlpxConnection = context.actorOf(RLPxConnectionHandler.props(nodeInfo), "rlpx-connection")
+    context watch rlpxConnection
+    rlpxConnection
+  }
+
+  def handleTerminated: Receive = {
+    case _: Terminated => context stop self
+  }
+
+  def waitingForConnectionResult(rlpxConnection: ActorRef): Receive = handleTerminated orElse {
+    case RLPxConnectionHandler.ConnectionEstablished =>
+      val hello = Hello(
+        p2pVersion = P2pVersion,
+        clientId = "etc-client",
+        capabilities = Seq(Capability("eth", EthProtocolVersion)),
+        listenPort = nodeInfo.listenAddress.getPort,
+        nodeId = ByteString(nodeInfo.nodeId))
+      rlpxConnection ! RLPxConnectionHandler.SendMessage(hello)
+      val timeout = system.scheduler.scheduleOnce(3.seconds, self, ProtocolHandshakeTimeout)
+      context become waitingForHello(rlpxConnection, timeout)
+
+    case RLPxConnectionHandler.ConnectionFailed =>
+      context stop self
+  }
+
+  def waitingForHello(rlpxConnection: ActorRef, timeout: Cancellable): Receive = handleTerminated orElse {
+    case MessageReceived(hello: Hello) =>
```

Comment:
@LukasGasior1 I think i'm receiving a `Disconnect(Too many peers) from peer` here and `case MessageReceived(d: Disconnect)` is missing. I haven't checked the protocol yet to figure out if a Disconnect is valid here (and not sure if it's a buggy peer).
The node I'm trying to connect to (in case you want to try it out) is `enode://65dfc3d6b1685a23c7801f89da0821f1ff6702f6380f5867c9918a1e0a34573da66b5e914ce91d719324beb6f32f2fcb98a858870dd9c34da72221d9ddf27793@77.72.144.197:30303`